wandb/sdk/internal/internal_api.py

```diff
@@ -1552,6 +1549,32 @@ def download_write_file(self, metadata, out_dir=None):
 
         return path, response
 
+    def upload_file_azure(self, url, file, extra_headers):
+        from azure.core.exceptions import HttpResponseError
+
+        client = self._azure_blob_module.BlobClient.from_blob_url(url)
+        try:
+            if extra_headers.get("Content-MD5") is not None:
+                md5 = base64.b64decode(extra_headers["Content-MD5"])
+            else:
+                md5 = None
+            content_settings = self._azure_blob_module.ContentSettings(
+                content_md5=md5, content_type=extra_headers.get("Content-Type"),
+            )
+            client.upload_blob(
+                file,
+                max_concurrency=4,
+                length=len(file),
+                overwrite=True,
+                content_settings=content_settings,
+            )
+        except HttpResponseError as e:
```

Comment:
Mind adding a test case for errors? Makes Codecov-kun happy :)

Reply:
I thought I already had with this one: https://github.com/wandb/client/pull/3089/files#diff-f7d8bbe325179a06284062672cfa16922802372cc4985e1f57c08366467e0651R50

I'll look into what's up.

Reply:
I see, yea, Codecov-kun _should_ be happy... I have noticed it missing lines that I'm 100% sure were covered in my latest PR, will read up on why this could be happening.

Reply:
Turns out codecov was right.  I needed to fix my test logic, new commit coming!